service/src/lib.rs

```diff
@@ -0,0 +1,23 @@
+#![feature(once_cell)]
+
+pub mod prove;
+pub mod verify;
+
+pub(crate) mod utils {
+    use std::ffi::{CStr, CString};
```

Comment:
CStr are truncated at '\x00'? So we must be careful here and recheck whether these codes are correct....

Reply:
About mem leak, we free the memory in `go-roller` and `scroll` by `C.free(xxx)`. 
``CStr are truncated at '\x00'?``
I do not see it in the source code of `Cstr::from_ptr()`. You can download code and click this function to jump to the location of the source code. ( The link I cannot find it through IDEA.

Reply:
after removing "serive" we don;t have Cstr here any more